controllers/gce/controller/utils.go

```diff
@@ -529,80 +587,39 @@ func (t *GCETranslator) getHTTPProbe(svc api_v1.Service, targetPort intstr.IntOr
 
 // isSimpleHTTPProbe returns true if the given Probe is:
 // - an HTTPGet probe, as opposed to a tcp or exec probe
-// - has a scheme of HTTP, as opposed to HTTPS
 // - has no special host or headers fields
 func isSimpleHTTPProbe(probe *api_v1.Probe) bool {
 	return (probe != nil && probe.Handler.HTTPGet != nil && probe.Handler.HTTPGet.Host == "" &&
-		probe.Handler.HTTPGet.Scheme == api_v1.URISchemeHTTP && len(probe.Handler.HTTPGet.HTTPHeaders) == 0)
+		len(probe.Handler.HTTPGet.HTTPHeaders) == 0)
 }
 
-// HealthCheck returns the http readiness probe for the endpoint backing the
-// given nodePort. If no probe is found it returns a health check with "" as
-// the request path, callers are responsible for swapping this out for the
-// appropriate default.
-func (t *GCETranslator) HealthCheck(port int64) (*compute.HttpHealthCheck, error) {
+// GetProbe returns a probe that's used for the given nodeport
+func (t *GCETranslator) GetProbe(port backends.ServicePort) (*api_v1.Probe, error) {
 	sl := t.svcLister.List()
-	var ingresses []extensions.Ingress
-	var healthCheck *compute.HttpHealthCheck
+
 	// Find the label and target port of the one service with the given nodePort
+	var service api_v1.Service
+	var svcPort api_v1.ServicePort
+	var found bool
+OuterLoop:
 	for _, as := range sl {
-		s := as.(*api_v1.Service)
-		for _, p := range s.Spec.Ports {
-
+		service = *as.(*api_v1.Service)
+		for _, sp := range service.Spec.Ports {
+			svcPort = sp
```

Comment:
Let me explain my thought process on this styling choice.  Ideally, `svcPort` is a pointer so nil can represent not found. However, `service.Spec.Ports[N]` is a struct which is why `svcPort` is a struct. You cannot take the address of `sp` because it's changing with every loop iteration. The other options are: 
1. Declare a new struct for each iteration and instantiate with value of sp, then take address of that variable if port is matched. Then you can check the pointer for existence, but you're not saving any lines of codes. 
2. Move `svcPort=sp` into `if` and check if ServicePort is a zero-value struct.  Of course, `if svcPort.Port != 0` is a bit of an eyesore. 

IMO, the current implementation and first alternative are equally fine and a matter of preference.